File: wrappers/java/src/test/java/org/hyperledger/indy/sdk/signus/SetDidMetadataTest.java
package org.hyperledger.indy.sdk.signus;

import org.hyperledger.indy.sdk.IndyIntegrationTestWithSingleWallet;
import org.hyperledger.indy.sdk.InvalidStructureException;
import org.junit.Test;

import java.util.concurrent.ExecutionException;

import static org.hamcrest.CoreMatchers.isA;
import static org.junit.Assert.assertEquals;


public class SetDidMetadataTest extends IndyIntegrationTestWithSingleWallet {

	@Test
	public void testSetDidMetadataWorks() throws Exception {
		Signus.setDidMetadata(wallet, DID, METADATA).get();
	}

	@Test
	public void testSetDidMetadataWorksForReplace() throws Exception {
		Signus.setDidMetadata(wallet, DID, METADATA).get();
		String receivedMetadata = Signus.getDidMetadata(wallet, DID).get();
		assertEquals(METADATA, receivedMetadata);

		String newMetadata = "updated metadata";
		Signus.setDidMetadata(wallet, DID, newMetadata).get();
		String updatedMetadata = Signus.getDidMetadata(wallet, DID).get();
		assertEquals(newMetadata, updatedMetadata);
	}

	@Test
	public void testSetDidMetadataWorksForEmptyString() throws Exception {
		Signus.setDidMetadata(wallet, DID, "").get();
	}

	@Test
	public void testSetDidMetadataWorksForInvalidDid() throws Exception {
		thrown.expect(ExecutionException.class);
		thrown.expectCause(isA(InvalidStructureException.class));

		Signus.setDidMetadata(wallet, INVALID_DID, METADATA).get();
	}
}

File: wrappers/dotnet/indy-sdk-dotnet-test/SignusTests/SetDidMetadataTest.cs
using Hyperledger.Indy.SignusApi;
using Microsoft.VisualStudio.TestTools.UnitTesting;
using System.Threading.Tasks;

namespace Hyperledger.Indy.Test.SignusTests
{
    [TestClass]
    public class SetDidMetadataTest : IndyIntegrationTestWithSingleWallet
    {
        [TestMethod]
        public async Task TestSetDidMetadataWorks()
        {
            await Signus.SetDidMetadataAsync(wallet, DID1, METADATA);
        }

        [TestMethod]
        public async Task TestSetDidMetadataWorksForReplace()
        {
            await Signus.SetDidMetadataAsync(wallet, DID1, METADATA);
            var receivedMetadata = await Signus.GetDidMetadataAsync(wallet, DID1);
            Assert.AreEqual(METADATA, receivedMetadata);

            var newMetadata = "updated metadata";
            await Signus.SetDidMetadataAsync(wallet, DID1, newMetadata);

            var updatedMetadata = await Signus.GetDidMetadataAsync(wallet, DID1);
            Assert.AreEqual(newMetadata, updatedMetadata);
        }

        [TestMethod]
        public async Task TestSetDidMetadataWorksForEmptyString()
        {
            await Signus.SetDidMetadataAsync(wallet, DID1, string.Empty);
        }

        [TestMethod]
        public async Task TestSetDidMetadataWorksForInvalidDid()
        {
            var ex = await Assert.ThrowsExceptionAsync<InvalidStructureException>(() =>
               Signus.SetDidMetadataAsync(wallet, "invalid_base58string", METADATA)
           );
        }
    }
}
